compiler/K_lift.fx

```diff
@@ -834,16 +908,32 @@ fun lift_all(kmods: kmodule_t list)
         kcb_kexp=Some(walk_kexp_n_lift_all)
     }
 
+    fun clear_unnecessary_closures(): void
+    {
+        for declj <- declosured_functions.list() {
+            match kinfo_(declj,noloc){  //TODO: Is noloc okay in this case, or we have to save locations strictly in map instead of set? (I'd better save reference than)
+                | KFun ((ref {kf_closure, kf_flags}) as kf) => 
+                    *kf = kf->{kf_closure=kf_closure.{kci_arg=noid, kci_fcv_t=noid, kci_make_fp=noid}, kf_flags=kf_flags.{fun_flag_uses_fv=false} }
+                | _ => throw compile_err(noloc,
+                    f"declosured functions list contains not a function '{idk2str(declj, noloc)}'")
+            }
+        }
+    }
+
     [: for km <- kmods {
         val {km_idx, km_top} = km
         val new_top = make_wrappers_for_nothrow(km_idx, km_top)
         curr_m_idx = km_idx
         curr_top_code = []
+        functions_met_as_vals.clear()
+        declosured_functions.clear()
         for e <- new_top { fold_defcl_kexp_(e, defcl_callb) }
+        declosured_functions.filter(fun (fn){!functions_met_as_vals.mem(fn)}) // TODO: Is there better way to find set subtraction?
```

Comment:
You are welcome to add the following function to Hashset.fx (e.g. immediately after `t.intersect(...)`):

```
fun t.diff(b: 'k Hashset.t): void
{
    val table = b.table
    for j <- 0:b.tabsz {
        val {hv, key} = table[j]
        if hv < HASH_SIGN_MASK {
            val (j, tidx) = self.find_idx_(key, hv)
            if tidx >= 0 {
                self.index.set(j, HASH_DELETED)
                self.table[tidx].hv = uint64(self.free) | HASH_SIGN_MASK
                self.free = tidx+1
                self.nactive -= 1
            }
        }
    }
}
```

Reply:
done